YAXLib/Pooling/PoolSettings.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (C) Sina Iravanian, Julian Verdurmen, axuno gGmbH and other contributors.
+// Licensed under the MIT license.
+
+#nullable enable
+
+namespace YAXLib.Pooling
+{
+    internal static class PoolSettings
+    {
+        public static bool CheckReturnedObjectsExistInPool { get; set; } = false;
```

Comment:
This test is expensive. Currently enabled in unit tests.

Reply:
Could we make it less expensive?

Reply:
Yes, consequently implementing `PooledObject<T> SpecializedPoolAbstract.PooledObject<T> Get(out T instance)`. The `PooledObject` will auto-return objects to the pool when disposing exactly once. 
What is left are coding errors in case of `T SpecializedPoolAbstract.Get()` and `SpecializedPoolAbstract.Return(T toReturn)`, which was never required by now. That's why `CheckReturnedObjectsExistInPool` is enabled only for debugging. 
So for the idea behind.